helm-charts/splunk-otel-collector/templates/_helpers.tpl

```diff
@@ -149,7 +149,12 @@ Define name for the Splunk Secret
 {{- if .Values.secret.name -}}
 {{- printf "%s" .Values.secret.name -}}
 {{- else -}}
-{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
+{{ $name := default .Chart.Name .Values.nameOverride -}}
```

Comment:
This logic is already defined in the template `splunk-otel-collector.fullname`. I would suggest just using the existing template here instead of c+p the same logic.